sherpa/optmethods/optfcts.py

```diff
@@ -529,6 +529,9 @@ def montecarlo(fcn, x0, xmin, xmax, ftol=EPSILON, maxfev=None, verbose=0,
     the pool of population, the resulting trial vector is selected if
     it lowers the objective function.
 
+    .. versionchanged:: 4.15.1
```

Comment:
update version number

Reply:
However, there is already support for using a RNG when `seed=None` so it looks like we would ideally want to allow a `rng` parameter...

I'm just slightly concerned as it ends up leaking through to the method options, and I'm not sure about what would happen if set the method option to a generator:

```
>>> from sherpa.astro import ui
>>> ui.get_method_name()
'levmar'
>>> ui.get_method_opt()
{'ftol': 1.1920928955078125e-07, 'xtol': 1.1920928955078125e-07, 'gtol': 1.1920928955078125e-07, 'maxfev': None, 'epsfcn': 1.1920928955078125e-07, 'factor': 100.0, 'numcores': 1, 'verbose': 0}
>>> ui.set_method("moncar")
>>> ui.get_method_name()
'moncar'
```

Note that the following has a `rng` field:

```
>>> ui.get_method_opt()
{'ftol': 1.1920928955078125e-07, 'maxfev': None, 'verbose': 0, 'seed': 74815, 'population_size': None, 'xprob': 0.9, 'weighting_factor': 0.8, 'numcores': 1, 'rng': None}
```

Reply:
Note: the version number has now been updated to `4.16.0` but the rest remains a valid question.